backend/test_observer/data_access/models.py

```diff
@@ -149,7 +154,15 @@ def architectures(self) -> set[str]:
             "name",
             "version",
             "track",
-            postgresql_where=column("track") != "",
+            postgresql_where=column("family") == FamilyName.snap.name,
+            unique=True,
+        ),
+        Index(
+            "unique_charm",
+            "name",
+            "version",
+            "track",
+            postgresql_where=column("family") == FamilyName.charm.name,
```

Comment:
This is nice, I was trying to do this when adding charms but can't remember why it was causing me grief, something to do with trying to join another table. Was family moved?